### PR TITLE
companies crud using popup and ajax

### PR DESCRIPTION
**Pull request by PragatiDoshi-BTC**
Company add and edit using popup and ajax

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -13,4 +13,13 @@ def configure_permitted_parameters
     devise_parameter_sanitizer.for(:account_update) { |u| u.permit(:email, :password, :password_confirmation, :first_name, :last_name, :current_password) }
   end
 
+  def set_company
+    if !current_user.nil?
```
Comment (amitpatelx) [inline app/controllers/application_controller.rb]: **bad**
`if !current_user.nil?`

**good**
`if current_user.present?`

Refer https://github.com/bbatsov/ruby-style-guide

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -13,4 +13,13 @@ def configure_permitted_parameters
...
+      if current_user.company.present? && current_user.role.to_s == "Admin"
```
Comment (amitpatelx) [inline app/controllers/application_controller.rb]: Use single quote for stirng literals

File: app/controllers/companies_controller.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+class CompaniesController < ApplicationController
```
Comment (amitpatelx) [inline app/controllers/companies_controller.rb:1]: Check entire class. Indentation is is not done properly

Refer https://github.com/bbatsov/ruby-style-guide#spaces-indentation

File: app/controllers/companies_controller.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+	before_action :authenticate_user!
+
+	def new
+
+	end
+
+	def create
```
Comment (amitpatelx) [inline app/controllers/companies_controller.rb]: This method is doing lots of thing that I don't understand. It simply doesn't follow SRP. Refactor it.

File: app/views/companies/create.js.haml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+- if flash[:error] != nil
```
Comment (amitpatelx) [inline app/views/companies/create.js.haml]: Indentation issue in whole file

File: app/views/companies/update.js.haml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+- if flash[:error] != nil
```
Comment (amitpatelx) [inline app/views/companies/update.js.haml]: indentation issues

File: app/views/layouts/application.html.haml [hunk 7]
```diff
@@ -19,6 +19,9 @@
               = link_to "Invoices", ""
             %li{class: "#{top_navbar_active_class_name('clients','index')}"} 
               = link_to "Clients", clients_path
+            -if current_user.present? && current_user.role.to_s == "Admin"
```
Comment (amitpatelx) [inline app/views/layouts/application.html.haml]: move this condition to a helper as a method

```
def is_admin?
  current_user.present? && current_user.role.to_s == "Admin"
end
```

and call this method wherever required

File: app/views/layouts/application.html.haml [hunk 8]
```diff
@@ -19,6 +19,9 @@
...
+              %li{class: "#{top_navbar_active_class_name('companies','new')}"} 
+                = link_to('Companies',"#modal-form","data-toggle" => "modal")
           %ul.nav.navbar-nav.pull-right
             -if !current_user.blank?
```
Comment (amitpatelx) [inline app/views/layouts/application.html.haml]: This is bad style again

Avoid using `!something` if possible

check http://api.rubyonrails.org/classes/Object.html#method-i-present-3F

File: app/views/layouts/application.html.haml [hunk 9]
```diff
@@ -33,3 +36,33 @@
     .container
       = render "layouts/shared/notification"
       = yield
+      -if current_user.present? && current_user.role.to_s == "Admin"
```
Comment (amitpatelx) [inline app/views/layouts/application.html.haml]: As mentioned earlier move, call the method as suggested.

File: app/views/layouts/application.html.haml [hunk 10]
```diff
@@ -33,3 +36,33 @@
...
+        = form_for @company, remote: true do |f|
```
Comment (amitpatelx) [inline app/views/layouts/application.html.haml]: Why this form exist in application layout?

Read this http://railsapps.github.io/rails-default-application-layout.html and http://guides.rubyonrails.org/layouts_and_rendering.html

File: config/routes.rb [hunk 11]
```diff
@@ -3,7 +3,7 @@
   get 'home/index'
```
Comment (amitpatelx) [inline config/routes.rb]: remove extra lines

File: app/controllers/companies_controller.rb [hunk 12]
```diff
@@ -0,0 +1,45 @@
+class CompaniesController < ApplicationController
+
+	before_action :authenticate_user!
+
+	def new
```
Comment (amitpatelx) [inline app/controllers/companies_controller.rb]: remove this method as it doesn't have any content

**Comment by amitpatelx** [comment]
@PragatiDoshi-BTC I don't see consistency in the condition checking for `nil` across the application

File: app/views/layouts/application.html.haml (hunk 10 above)
Comment (PragatiDoshi-BTC) [inline app/views/layouts/application.html.haml]: This form should be available to all the pages, as it is a link in the navigation bar, so this form is in application layout.

File: app/controllers/companies_controller.rb (hunk 4 above)
Comment (PragatiDoshi-BTC) [inline app/controllers/companies_controller.rb]: This method will find admin_id for the users, i.e. for particular user, which user is admin for that. I will remove some unnecessary code. Please review and suggest.

File: app/views/layouts/application.html.haml (hunk 10 above)
Comment (amitpatelx) [inline app/views/layouts/application.html.haml]: Then move it to a partial with proper name

File: app/controllers/companies_controller.rb (hunk 4 above)
Comment (amitpatelx) [inline app/controllers/companies_controller.rb]: First make necessary changes and proper indentation, commit changes.

**Pull request merged**